src/cluster.ml

```diff
@@ -0,0 +1,213 @@
+open Current.Syntax
+
+module Arch = struct
+  type t = [
+    | `Linux_arm64
+    | `Linux_x86_64
+    | `Linux_ppc64
+    | `Linux_s390x
+    | `Linux_riscv64
+    ]
+
+  let pool_id : t -> string = function
+    | `Linux_arm64 -> "linux-arm64"
+    | `Linux_x86_64 -> "linux-x86_64"
+    | `Linux_ppc64 -> "linux-ppc64"
+    | `Linux_s390x -> "linux-s390x"
+    | `Linux_riscv64 -> "linux-riscv64"
+end
+
+let password_path = "/run/secrets/ocurrent-hub"
+let push_repo = "ocurrentbuilder/staging"
+
+let auth =
+  if Sys.file_exists password_path then (
+    let ch = open_in_bin password_path in
+    let len = in_channel_length ch in
+    let password = really_input_string ch len |> String.trim in
+    close_in ch;
+    Some ("ocurrent", password)
+  ) else (
+    Fmt.pr "Password file %S not found; images will not be pushed to hub@." password_path;
+    None
+  )
+
+(* Strings here represent the docker context to use. *)
+module Ci3_docker = Current_docker.Default
+module Ci4_docker = Current_docker.Make(struct let docker_context = Some "ci4.ocamllabs.io" end)
+module Docs_docker = Current_docker.Make(struct let docker_context = Some "docs.ci.ocaml.org" end)
+module Staging_docs_docker = Current_docker.Make(struct let docker_context = Some "staging.docs.ci.ocamllabs.io" end)
+module Ci_docker = Current_docker.Make(struct let docker_context = Some "ocaml.ci.dev" end)
+module Opamrepo_docker = Current_docker.Make(struct let docker_context = Some "opam.ci.ocaml.org" end)
+module Check_docker = Current_docker.Make(struct let docker_context = Some "check.ci.ocaml.org" end)
+module Watch_docker = Current_docker.Make(struct let docker_context = Some "watch.ocaml.org" end)
+module Ocamlorg_docker = Current_docker.Make(struct let docker_context = Some "ocaml-www1" end)
+module Cimirage_docker = Current_docker.Make(struct let docker_context = Some "ci.mirage.io" end)
+module V2ocamlorg_docker = Current_docker.Make(struct let docker_context = Some "v2.ocaml.org" end)
+module Ocamlorg_images = Current_docker.Make(struct let docker_context = Some "ci3.ocamllabs.io" end)
+module Docker_aws = Current_docker.Make(struct let docker_context = Some "awsecs" end)
+module V3b_docker = Current_docker.Make(struct let docker_context = Some "v3b.ocaml.org" end)
+module V3c_docker = Current_docker.Make(struct let docker_context = Some "v3c.ocaml.org" end)
+module Deploycamlorg_docker = Current_docker.Default
```

Comment:
This is not a comment that needs to be (nor should be) addressed in this PR, but I'm just noticing that `Deploycamlorg` and `Ci3` are the same module. I wonder what is the value in  giving them a name here...